cachi2/core/package_managers/rpm/main.py

```diff
@@ -0,0 +1,204 @@
+import logging
+import os
+import shlex
+from pathlib import Path
+from typing import Any
+from urllib.parse import quote
+
+import rpmfile
+import yaml
+
+from cachi2.core.errors import PackageRejected
+from cachi2.core.models.input import Request
+from cachi2.core.models.output import EnvironmentVariable, ProjectFile, RequestOutput
+from cachi2.core.models.sbom import Component
+from cachi2.core.package_managers.rpm.base import RpmsLockT
+from cachi2.core.package_managers.rpm.redhat import RedhatRpmsLock
+from cachi2.core.rooted_path import RootedPath
+from cachi2.core.utils import run_cmd
+
+log = logging.getLogger(__name__)
+
+
+DEFAULT_LOCKFILE_NAME = "rpms.lock.yaml"
+DEFAULT_PACKAGE_DIR = "deps/rpm"
+
+
+class LockfileFormat:
+    """The class will handle all lockfile formats."""
+
+    def __init__(self, raw_content: dict[Any, Any]) -> None:
+        self._handlers = []
+        self._raw_content = raw_content
+        self._matched_handler = None
+
+    @property
+    def matched_handler(self) -> RpmsLockT:
+        return self._matched_handler
+
+    def add_handler(self, handler) -> None:
+        self._handlers.append(handler(self._raw_content))
+
+    def process_formats(self) -> None:
+        for handler in self._handlers:
+            log.debug("Checking format '{}'".format(handler.__class__.__name__))
+            if handler.match_format():
+                log.debug("Lockfile content validation")
+                handler.process_format()
+                if handler.is_valid():
+                    self._matched_handler = handler
+                    return
+
+        raise PackageRejected(
+            f"Rpm lockfile '{DEFAULT_LOCKFILE_NAME}' format is not supported'.",
+            solution=("Check 'lockfileVendor' and 'lockfileVersion' keys in the lockfile."),
+        )
+
+
+def fetch_rpm_source(request: Request) -> RequestOutput:
+    """Process the input lockfile, fetch packages and generate SBOM."""
+    components: list[Component] = []
+
+    _check_lockfile(request)
+    lockfile_name = request.source_dir.join_within_root(DEFAULT_LOCKFILE_NAME)
```

Comment:
We're assuming that the lockfile will be present in the root of the repository, and also that there'll be a single one of them. But Cachi2's design allows the processing of multiple [input packages](https://github.com/containerbuildsystem/cachi2/blob/main/cachi2/core/package_managers/yarn/main.py#L26-L30) (think of them as individual projects, such as a folder that is a Python project and contains a `requirements.txt` file), and also allows the user the define the relative path to such package.

See how this is done in [yarn](https://github.com/containerbuildsystem/cachi2/blob/main/cachi2/core/package_managers/yarn/main.py#L26-L30). I assume we want to follow the same idea for RPMs.

Reply:
Here's a draft of an integration test to cover this scenario: https://github.com/containerbuildsystem/cachi2/commit/cfe3813b9b0f885d02fab5adfa685d049e2b56b2